### ISSUE TITLE
[HOLD for payment 2024-08-01] [$125] [Payment card / Subscription] Make payment card only display last four digits

### ISSUE THREAD

**Issue by blimpich** [Expensify/App#45309]
# Problem

Right now when you add a payment card to the new subscription page, it shows the 16 characters instead of just four:

![Image](https://github.com/Expensify/App/assets/17171674/5100bfcd-54e2-47c6-a9a3-cbe48c26bc78)

It should show just the last four:

![Image](https://github.com/Expensify/App/assets/17171674/a90f0b06-7481-484a-8c93-ad6f5a335791)

# Solution

Lets fix this by only showing the last four digits in the subscription page.
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01a9042029c460e7a0</li>
        <li>Upwork Job ID: 1811455490190961117</li>
        <li>Last Price Increase: 2024-07-11</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @CortneyOfstad / @isabelastisser</details>

**Comment by blimpich** [comment]
Lowering price because this is pretty straightforward string manipulation.

**Comment by blimpich** [comment]
Anyone who tests this should be able to simulate adding a test card by populating the `fundList` onyx key based off the types we define in the frontend, or you can add a real credit card, though know that we currently don't have a way of deleting credit cards added to the app.

**Comment by kabeer95** [comment]
Feature Request: Mask Payment Card Numbers on Expensify

Summary:

To enhance security and compliance, we propose masking payment card numbers on Expensify, displaying only the last four digits. This feature will provide an additional layer of protection for sensitive payment information.

Current State:

Currently, Expensify displays the full payment card number, which may pose a security risk if unauthorized access is gained.

Proposed Solution:

Implement a feature to mask payment card numbers, displaying only the last four digits. For example:

Original card number: 424242XXXXXX4242
Masked card number: ************4242
Benefits:

Enhanced Security: Masking payment card numbers reduces the risk of unauthorized access to sensitive information.
Compliance: This feature aligns with industry standards for payment card security, such as PCI-DSS.
User Trust: By protecting payment information, Expensify demonstrates a commitment to user security and trust.
Implementation:

To implement this feature, we recommend the following:

Update the payment card number display to show only the last four digits.
Store the full payment card number securely, using industry-standard encryption and access controls.
Ensure that the masked card number is displayed consistently throughout the Expensify platform.
Estimated Development Time:

We estimate 20 hours to implement this feature and mask payment card numbers on Expensify.

Priority:

We consider this feature a high priority, as it directly impacts the security and trust of Expensify users.

Please let us know if you would like to discuss this feature further or proceed with implementation

**Comment by sobitneupane** [comment]
Thanks for the proposal everyone.

[Proposal](https://github.com/Expensify/App/issues/45309#issuecomment-2226781149) from @tienifr looks good to me.

🎀 👀 🎀 C+ reviewed

**Comment by blimpich** [comment]
Yes that should be fine 👍

**Comment by isabelastisser** [comment]
Hi @sobitneupane, PR is ready for review. Thanks!

**Comment by isabelastisser** [comment]
I will be OOO tomorrow and next week, so I am reassigning this until I return on July 29. Thanks, @CortneyOfstad!

Status: Waiting for PR review.

**Comment by isabelastisser** [comment]
I'm back and taking this back. Thanks, @CortneyOfstad !

**Comment by isabelastisser** [comment]
Not overdue.

**Comment by isabelastisser** [comment]
Hey @sobitneupane, please complete the BZ list above. Thanks!

**Comment by sobitneupane** [comment]
> **BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:
> 
> * [ ]  [@sobitneupane] The PR that introduced the bug has been identified. Link to the PR:

https://github.com/Expensify/App/pull/44072
> * [ ]  [@sobitneupane] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:

https://github.com/Expensify/App/pull/44072#issuecomment-2268631759
> * [ ]  [@sobitneupane] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:

Not required.
> * [ ]  [@sobitneupane] Determine if we should create a regression test for this bug.

Yes.
> * [ ]  [@sobitneupane] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.

**Comment by sobitneupane** [comment]
**Regression Test Proposal:**

- Go to /settings/subscription.
- Add a new card if there is no.
- Verify card title displays only last four digits of your card.

Do we agree 👍 or 👎

**Comment by isabelastisser** [comment]
Payment summary:

C+ review @sobitneupane owed $250 via NewDot
Contributor @tienifr owed $250 via NewDot

**Comment by isabelastisser** [comment]
All set.

**Comment by garrettmknight** [comment]
$250 approved for @sobitneupane based on [this summary.](https://github.com/Expensify/App/issues/45309#issuecomment-2269375290)

**Comment by garrettmknight** [comment]
$250 approved for @tienifr
